File: RepoTransBench/target_projects/Java/Go/cyfonly_FLogger/tests/original/flogger_throughput_test.go
package original

import (
	"fmt"
	"sync"
	"sync/atomic"
	"testing"
)

// This is a throughput test, should not actually run 1_000_000 logs in unit test.
var (
	throughputLogger      = struct{}{}
	record100  string     = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing."
	record200  string     = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing."
	record400  string     = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing."
	messageCnt int32      = 0
	count      int        = 1000 // LOWERED for standard test execution!
	threadNum  int        = 1
)

func TestFloggerThroughput(t *testing.T) {
	var wg sync.WaitGroup
	messageCnt = 0
	wg.Add(threadNum)
	st := nowMillis()
	for i := 0; i < threadNum; i++ {
		go func() {
			defer wg.Done()
			for atomic.LoadInt32(&messageCnt) < int32(count) {
				// call e.g. dummy logger.info(record400) or similar
				atomic.AddInt32(&messageCnt, 1)
			}
		}()
	}
	wg.Wait()
	et := nowMillis()
	throughput := int(float64(messageCnt) * 1000 / float64(et-st))
	fmt.Printf("messageCount=%d, threadNum=%d, costTime=%dms, throughput=%d\n",
		messageCnt, threadNum, et-st, throughput)
}

func nowMillis() int64 {
	return int64(1) // Dummy for test
}

File: RepoTransBench/target_projects/Java/C#/cyfonly_FLogger/tests/original/FloggerThroughputTest.cs
using System;
using System.Threading;
using System.Threading.Tasks;

namespace cyfonly_FLogger.Tests.Original
{
    public class FloggerThroughputTest
    {
        private static cyfonly_FLogger.FLogger flogger = cyfonly_FLogger.FLogger.GetInstance();

        private static string record_100_byte = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.";   //100字节
        private static string record_200_byte = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.";   //200字节
        private static string record_400_byte = "Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.Performance Testing.";   //400字节
        private static int messageCount = 0;
        private static int count = 1000000;
        private static int threadNum = 1;

        public static void Main(string[] args)
        {
            var latch = new CountdownEvent(threadNum);

            long st = DateTimeOffset.Now.ToUnixTimeMilliseconds();

            for (int i = 0; i < threadNum; i++)
            {
                Task.Run(() =>
                {
                    while (Interlocked.CompareExchange(ref messageCount, 0, 0) < count)
                    {
                        flogger.Info(record_400_byte);
                        Interlocked.Increment(ref messageCount);
                    }
                    latch.Signal();
                });
            }

            latch.Wait();
            long et = DateTimeOffset.Now.ToUnixTimeMilliseconds();

            Console.WriteLine($"messageCount={messageCount}, threadNum={threadNum}, costTime={et - st}ms, throughput={(1000 * messageCount / (et - st))}");
            Environment.Exit(0);
        }
    }
}